Shell/main.cpp

```diff
@@ -300,16 +300,41 @@ static Vector<String> expand_globs(const StringView& path, const StringView& bas
     return res;
 }
 
+static Vector<String> expand_parameters(const StringView& param)
+{
+    bool is_variable = param.length() > 1 && param[0] == '$';
+    if (!is_variable)
+        return { param };
+
+    String param_name = String(param.substring_view(1, param.length() - 1));
```

Comment:
This is a variable name, so let's call it `variable_name`.
Also, you can use implicit conversion:
```suggestion
String variable_name = param.substring_view(1, param.length() - 1);
```